src/content/docs/en/guides/backend/xata.mdx

```diff
@@ -6,9 +6,142 @@ service: Xata
+:::
+
+Once you follow the prompt, you should have a `.env` file in the root of your project with the following variables defined:
+
+```ini title=".env"
+XATA_API_KEY=hash_key
+
+# Xata branch that will be used
+# if there's not a xata branch with
+# the same name as your git branch
+XATA_FALLBACK_BRANCH=main
+```
+
+To have IntelliSense for your environment variables, edit or create the file `env.d.ts` in your `/src` directory to make your environment variables type-safe:
+
+```ts title="src/env.d.ts"
+interface ImportMetaEnv {
+  readonly XATA_API_KEY: string;
+  readonly XATA_FALLBACK_BRANCH?: string;
+}
+
+interface ImportMeta {
+  readonly env: ImportMetaEnv;
+}
+```
+
+:::tip
+Read more about [environment variables](/en/guides/environment-variables/) and `.env` files in Astro.
+:::
+
+And a `.xatarc` file with a `databaseURL`. Though it can be extended to have more definitions, such as the output path for your codegen, e.g.:
+
+```json title=".xatarc"
+{
+  "databaseURL": "https://{{ Xata DB URL }}",
+  "codegen": {
+    "output": "src/xata.codegen.ts"
+  }
+}
+```
+
+Additionally, the prompt may have assisted you in creating an instance of the SDK (Software Development Kit) and, if you chose TypeScript, you also have the types generated from your schema. Notice `@xata.io/client` was added to your `package.json` as well.
+
+Finally, your project should have the following structure:
+
+<FileTree title="Project Structure">
+- src/
+  - **xata.codegen.ts**
+  - **env.d.ts**
+- **.env**
+- astro.config.mjs
+- package.json
+- **.xatarc**
+</FileTree>
+
+## Create your queries
+
+To illustrate how to make a query, let's query the first 50 posts from Xata's Sample Blog Database.
+
+```astro title="src/pages/blog/index.astro"
+---
+import { getXataClient } from "../xata.codegen";
+
+const xata = getXataClient();
+const { records } = await xata.db.Posts.getPaginated({
+  pagination: {
+    size: 50
+  }
+})
+---
+
+<ul>
+  {records.map(post) => (
+    <li>{post.title}</li>
+  )}
+</ul>
+```
+
+Since the SDK is generated dynamically, and will be updated any time you tweak your schema, it's better to leave it untouched. In case you wish to hoist a few common queries to keep your code base DRY, it's a best practice to create an additional file where your queries can be re-used.
+
+## Different query strategies
+
+As the Xata SDK connects directly to your database, it is important to never query from the client-side. As a matter of fact, the SDK access `process.env` and thus it will not even work.
+
+:::caution
+Never import the Xata SDK into your client-side bundle.
+:::
+
+So there are 3 ways to fetch your data from Xata in an Astro app:
+
+1. On-build time, in a prerendered page.
+2. On-demand, in a [server-side rendered (SSR)](/en/guides/server-side-rendering/#enabling-ssr-in-your-project) page.
+3. From the client-side with a Backend-For-Frontend (BFF) in the middle, such as a Serverless Function.
```

Comment:
```suggestion
```
I'm also going to suggest we remove this as not immediately helpful. If you only show approved/recommended usage, then that's much better for the reader. We typically would not include sections that are mostly just a big caution, do not do this.

However, if there is a key difference using Xata with SSR than a pre-rendered page, I think an SSR section makes sense here! If you'd like to show an example of using Xata with SSR, that's perfectly appropriate!